pkg/rest/vf/vm_config.go

```diff
@@ -4,28 +4,30 @@ import (
 	"net/http"
 
 	"github.com/Code-Hex/vz/v3"
+	"github.com/crc-org/vfkit/pkg/config"
 	"github.com/crc-org/vfkit/pkg/rest/define"
 	"github.com/gin-gonic/gin"
 	"github.com/sirupsen/logrus"
 )
 
 type VzVirtualMachine struct {
-	VzVM   *vz.VirtualMachine
-	config *vz.VirtualMachineConfiguration
+	VzVM     *vz.VirtualMachine
+	config   *vz.VirtualMachineConfiguration
+	vmConfig *config.VirtualMachine
```

Comment:
That's getting confusing, too many configs/vms/... data types ^^
Not a problem with this PR, just thinking out loud. Maybe will be reworked another day in a refactor

Reply:
Perhaps we can use the built-in http package in golang to handle this🤔. We can create an issue for this (and maybe add the `good first issue` label?).

Reply:
I've started reworking this in this branch https://github.com/cfergeau/vfkit/commits/vzvm/ as it seemed to be a requirement for other changes I wanted to make, see the `rework types` commit.
The idea is to have a `VfVirtualMachineConfig` type which wraps `vz.VirtualMachineConfiguration` + `vfkit.config.VirtualMachine` and a `VfVirtualMachine` type which wraps `vz.VirtualMachine` and `VfVirtualMachineConfig`.

This way, with only a `VfVirtualMachine` instance, you have access to all the `vz` and `vf` related objects.